web/webkit/src/main/scala/net/liftweb/http/LiftSession.scala

```diff
@@ -579,12 +603,18 @@ class LiftSession(private[http] val _contextPath: String, val underlyingId: Stri
    * Updates the internal functions mapping
    */
   def updateFunctionMap(funcs: Map[String, S.AFuncHolder], uniqueId: String, when: Long): Unit = {
-    funcs.foreach {
+    copyFunctions(funcs, nmessageCallback)
+
+    if(LiftRules.putAjaxFnsInContainerSession)
+      updateHttpSessionFns(copyFunctions(funcs, _))
+  }
+
+  private def copyFunctions(from: Map[String, S.AFuncHolder], to:java.util.Map[String, S.AFuncHolder]):Unit =
+    from.foreach {
       case (name, func) =>
-        nmessageCallback.put(name,
+        to.put(name,
           if (func.owner == Full(uniqueId)) func else func.duplicate(uniqueId))
     }
```

Comment:
Er… That `put` is what ensures that `RequestVar` context will be available in AJAX callbacks. So if you're not managing to capture it somehow, your AJAX callbacks will be able to run elsewhere, but only without part of their context.